core/common/src/main/java/alluxio/retry/TimeBoundedExponentialRetry.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.retry;
+
+import alluxio.time.TimeContext;
+
+import java.time.Duration;
+
+/**
+ * A retry policy which uses exponential backoff and a maximum duration time bound.
+ *
+ * A final retry will be performed at the time bound before giving up.
+ *
+ * For example, with initial sleep 10ms, maximum sleep 100ms, and maximum duration 500ms, the sleep
+ * timings would be [10, 20, 40, 80, 100, 100, 100, 50], assuming the operation being retries takes
+ * no time. The 50 at the end is because the previous times add up to 450, so the mechanism sleeps
+ * for only 50ms before the final attempt.
+ */
+public final class TimeBoundedExponentialRetry extends TimeBoundedRetry {
```

Comment:
NIT: is `ExponentialBackoffTimeBoundedRetry` more consistent with our naming convention?

Reply:
Renamed to `ExponentialTimeBoundedRetry`. Removed `Backoff` since the name is getting too long and I think it should be clear enough what it means